File: src/test/scala/com/github/tetrisanalyzer/piecegenerator/PredictablePieceGeneratorTest.scala
package com.github.tetrisanalyzer.piecegenerator

import org.junit.Test
import com.github.tetrisanalyzer.BaseTest
import com.github.tetrisanalyzer.piece.{PieceO, PieceI, PieceZ, PieceS}

class PredictablePieceGeneratorTest extends BaseTest {

  @Test def nextPieceNumber {
    val pieceGenerator = new PredictablePieceGenerator(List(new PieceS, new PieceZ, new PieceI))
    val result = pieceGenerator.nextPieceNumber ::
      pieceGenerator.nextPieceNumber ::
      pieceGenerator.nextPieceNumber :: Nil;

    result should be (List(new PieceS().number, new PieceZ().number, new PieceI().number))
  }

  @Test def runOutOfPieces {
    val pieceGenerator = new PredictablePieceGenerator(List(new PieceI))
    val result = pieceGenerator.nextPieceNumber ::
      pieceGenerator.nextPieceNumber ::
      pieceGenerator.nextPieceNumber :: Nil;

    result should be (List(new PieceI().number, new PieceO().number, new PieceO().number))
  }
}

File: otherlang/src/test/java/com/github/tetrisanalyzer/piecegenerator/PredictablePieceGeneratorTest.java
package nu.tengstrand.tetrisanalyzer.piecegenerator;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class PredictablePieceGeneratorTest {

    @Test
    public void nextPieceNumber() {
        PieceGenerator pieceGenerator = new PredictablePieceGenerator(Arrays.asList(new PieceS(), new PieceZ(), new PieceI()));
        List<Integer> result = Arrays.asList(pieceGenerator.nextPieceNumber(),
            pieceGenerator.nextPieceNumber(),
            pieceGenerator.nextPieceNumber());

        assertEquals(Arrays.asList((int)new PieceS().number(), (int)new PieceZ().number(), (int)new PieceI().number()), result);
    }

    @Test
    public void runOutOfPieces() {
        PieceGenerator pieceGenerator = new PredictablePieceGenerator(Arrays.asList((Piece)new PieceI()));
        List<Integer> result = Arrays.asList(pieceGenerator.nextPieceNumber(),
            pieceGenerator.nextPieceNumber(),
            pieceGenerator.nextPieceNumber());

        assertEquals(Arrays.asList((int)new PieceI().number(), (int)new PieceO().number(), (int)new PieceO().number()), result);
    }
}
